src/odemis/gui/cont/correlation.py

```diff
@@ -233,6 +235,13 @@ def add_streams(self, streams: list) -> None:
             sc = self._tab.streambar_controller.addStream(s, add_to_view=True, play=False)
             sc.stream_panel.show_remove_btn(True)
 
+    def _stop_streams_subscriber(self):
+        self._tab_data_model.streams.unsubscribe(self.update_localization_tab_streams)
+
+    @call_in_wx_main
+    def _start_streams_subscriber(self):
+        self._tab_data_model.streams.subscribe(self.update_localization_tab_streams, init=False)
+
     def update_localization_tab_streams(self, streams: list) -> None:
```

Comment:
Most likely this function should be decorated by `@call_in_wx_main`. In particular, it calls `showOverviewStream()`, which states that it must be running in the GUI thread.